### ISSUE TITLE
Music control/Button settings on Gadgetbridge

### ISSUE THREAD

**Issue by cored4** [Freeyourgadget/Gadgetbridge#979]
Hi, before I start I wanted to say - Gadgetbridge is great. It's good to have open-source app for smartbands. It's pretty much perfect for me, but I think there's one or two things that would boost app functionality by a lot. What I mean is music control on button press and/or just button settings in general. I saw such thing once in app "Mi Bandage" and it works nice but I don't really want to use two apps where one is closed source. But it would be really nice to have configurable Mi Band 2 button on Gadgetbridge with something like Next/Previous song on tap or answer call with double tap. Not sure if it'd be possible but I'd really love to see such option in that app :)

**Comment by NotAFile** [comment]
You can already use the Mi Band Button to trigger an intent. And I've seen work on being to be able to trigger different intents based on different tap counts.

Of course, this doesn't directly control music, but the question is really if that makes sense,  since everyone might want to use it for something else.

**Comment by cored4** [comment]
Of course, that was pretty much just example, different button actions on multiple taps would be great by itself. I heard it's possible to trigger an intent using button but I've never seen option to set that intent on Gadgetbridge, is it really there? Unless you mean just rejecting call with long press.

**Comment by danielegobbetti** [comment]
Yes the miband settings page contains a "button actions" sub-page since a few releases

**Comment by selurvedu** [comment]
I was able to perform an action (play/pause music) using Gadgetbridge, Mi Band 2 and Easer.

Here's how:

1. Install Easer, configure its permissions.
2. Create a new profile with any title (e.g. "PlayPause"), add the "Media Control" operation and tick the checkbox on the left, then select "Toggle Play/Pause" from the list. Save the profile.
3. Add a new script with any title (e.g. "Control playback with Gadgetbridge").
Leave the "Parent" field blank.
Select the previously created profile in the second field.
Select the "Inline Event" in the next line.
Leave "Reverse Scenario" disabled.
Swipe the list of available events to the right, select "Receive Broadcast".
Enter the following text into the "Action" field: `nodomain.freeyourgadget.gadgetbridge.mibandButtonPressed` (you can copy it from the "Broadcast message to send" setting in the button action settings of Gadgetbridge).
Leave the "Category" field blank.
Save the script.
4. Open the main screen of Easer and press "Start" in the three-dots menu.
5. Make sure that you are connected to your wristband, also make sure that the button action is enabled and properly configured in Gadgetbridge.
6. Start your music player (it also works if the player is already running in the background). Tap the button on your Mi Band until the music playback is started or stopped.

Please note:
1. Selecting "Inline Event" is important, because otherwise (if used with predefined "event" or "condition") the button action will only work once. At least that was my result, correct me if I'm wrong.
2. The name of the broadcast can be changed, but it must match in both Easer and Gadgetbridge.

**Comment by selurvedu** [comment]
@dj0001, I tried that on Mi Band 3 as well and it didn't work. See https://github.com/Freeyourgadget/Gadgetbridge/issues/1299#issuecomment-429363978 and #924.

P. S. Please avoid overquoting when possible, use at-mentions (@selurvedu).

**Comment by dj0001** [comment]
serlurvedu, thanks for commenting.
what tried you ' play/pause music using Gadgetbridge, Mi Band 3 and Easer ' or my webapp?
I logged the bluetooth traffic with https://github.com/vshymanskyy/miband-js/tree/gh-pages and there was no traffic, when pressing the big button. So I think, it don't work with the big button.

Selecting More on Miband3 and then device search(long press) generatates bluetooth traffic.
This is the way, my webapp works.
Break up gadgetgetbridge, open my website, select camera and connect bluetooth.
then go to devicesearch on miband, longpress, and theMediaDevice shots a picture, which can be saved.

**Comment by selurvedu** [comment]
@dj0001, I guess that works, but Gadgetbridge already responds to the activation of that menu item by opening the "Find Phone" screen: #1306

**Comment by spyjoshx** [comment]
Is there anything new with this? I think the big issue here is that button actions aren't recognized from Mi Band 3. (And yes, I know that it has to be one press, when the screen is off). Is this an easy fix, or is it something that would require a lot of work?

**Comment by selurvedu** [comment]
JFYI, the "Notify & Fitness for Mi Band" app (com.mc.miband1) supports both single button press and a (recommended) way of triggering the action – by initiating the heart rate measurement and immediately cancelling it (AFAIR, you have to do that three times it a row, though the number is most likely configurable).
